py/dml/ctree.py

```diff
@@ -4132,11 +4132,26 @@ def mkCast(site, expr, new_type):
         assert (not (real.is_int and real.is_endian)
                 and not (old_type.is_int and old_type.is_endian))
         return Cast(site, expr, new_type)
-    if ((isinstance(real, TPtr) or real.is_int or isinstance(real, TBool))
-        and (isinstance(old_type, (TPtr, TArray, TFunction))
-             or old_type.is_int or isinstance(old_type, TBool))):
+    if ((isinstance(real, (TBool, TPtr)) or real.is_int)
+        and (isinstance(old_type, (TBool, TPtr, TArray, TFunction))
+             or old_type.is_int)):
         assert (not (real.is_int and real.is_endian)
                 and not (old_type.is_int and old_type.is_endian))
+        if isinstance(old_type, (TPtr, TArray)) and isinstance(real, TPtr):
+            old_base = old_type.base
+            new_base = real.base
+            if (not dml.globals.compat_dml12_int(site)
+                and isinstance(old_base, (TLayout, TEndianInt))
+                and new_base.is_int and not new_base.is_endian
+                and not new_base.bits == 8):
```

Comment:
It seems far-fetched... but at that point since we're just using a single byte representation is guaranteed the same and so we don't get the strict aliasing violations anyway. So I might as well not warn in that case.

Reply:
actually wait it _might_, _technically_ violate strict aliasing. `uint8_le_t` is `struct { uint8 }`, not `uint8` directly. That means arrays of them aren't considered char arrays by the C standard, and so are more susceptible to strict aliasing violations.
Maybe?

Reply:
I'm not sure... and because of that I'm leaning towards not warning about it.